### PR TITLE
added component entity class and component type enum

### PR DESCRIPTION
**Pull request by andrlos**
addition of entity Component class and ComponentType enum

### CODE REVIEW

File: src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public void setType(ComponentType type) {
+        this.type = type;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java:66]: instanceOf :trollface:

File: src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        Component component = (Component) o;
+        return id == component.id && Objects.equals(car, component.car) && type == component.type;
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java:68]: gettery!

**Review by Jakub-Fajkus — CHANGES_REQUESTED** [review]

File: src/main/java/cz/muni/pa165/teamwhite/formula1/enums/ComponentType.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package cz.muni.pa165.teamwhite.formula1.enums;
+
+public enum ComponentType {
+    ENGINE,
+    SUSPENSION,
+    TRANSMISSION,
+    RIMS,
+    READSPOILER
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/enums/ComponentType.java:8]:
```suggestion
    REAR_SPOILER
```

**Review by Jakub-Fajkus — APPROVED** [review]

File: src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    private Car car;
+
+    @Column
+    @NotNull
+    private ComponentType type;
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java:29]:
```suggestion
    @Enumerated(EnumType.STRING)
    private ComponentType type;
```

**Review by Jakub-Fajkus — CHANGES_REQUESTED** [review]

File: src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
+package cz.muni.pa165.teamwhite.formula1.entity;
+
+import cz.muni.pa165.teamwhite.formula1.enums.ComponentType;
+
+import javax.persistence.*;
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java:5]: neimportovat vsechno prosim :trollface:

File: src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    private Car car;
+
+    @Column
+    @NotNull
+    @Enumerated(EnumType.STRING)
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java:24]: :+1:

File: src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+@Entity
+public class Component {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    private Car car;
+
+    @Column
```
Comment (Jakub-Fajkus) [inline src/main/java/cz/muni/pa165/teamwhite/formula1/entity/Component.java:22]: Musi byt column kdyz je enumerated?

**Review by Jakub-Fajkus — CHANGES_REQUESTED** [review]

**Review by Jakub-Fajkus — APPROVED** [review]

**Review by khecova — APPROVED** [review]
ok

**Comment by khecova** [comment]
I make DAO for this.

**Review by tomasedl — APPROVED** [review]
ok

**Pull request merged**